### PR TITLE
[AU4] Added beats&measures ruler

### PR DESCRIPTION
**Pull request by grliszas14**
Resolves: https://github.com/audacity/audacity/issues/6661

### CODE REVIEW

File: au4/src/projectscene/view/timeline/beatsmeasuresformat.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+#pragma once
+
+#include "view/timeline/irulerformat.h"
+
+namespace au::projectscene {
+class BeatsMeasuresFormat : public IRulerFormat
+{
+public:
+    IntervalInfo intervalInfo(TimelineContext* context);
+    QString label(double d, const IntervalInfo& timeIntervalInfo, TickType tickType, TimelineContext* context);
```
Comment (igorkorsukov) [inline au4/src/projectscene/view/timeline/beatsmeasuresformat.h:16]: need to add override

File: au4/src/projectscene/view/timeline/timeformat.h [hunk 2]
```diff
@@ -6,27 +6,13 @@
...
+class TimeFormat : public IRulerFormat
 {
 public:
-    explicit TimeFormat() {}
-    ~TimeFormat();
-
-    static TimeIntervalInfo timeIntervalInfo(double zoom);
-    static QString label(double d, const TimeIntervalInfo& timeIntervalInfo, TickType tickType);
+    IntervalInfo intervalInfo(TimelineContext* context);
+    QString label(double d, const IntervalInfo& intervalInfo, TickType tickType, TimelineContext* context = nullptr);
```
Comment (igorkorsukov) [inline au4/src/projectscene/view/timeline/timeformat.h:16]: need to add override

**Review by igorkorsukov — APPROVED** [review]

File: au4/src/projectscene/view/timeline/beatsmeasuresformat.h (hunk 1 above)
Comment (grliszas14) [reply au4/src/projectscene/view/timeline/beatsmeasuresformat.h]: added

File: au4/src/projectscene/view/timeline/timeformat.h (hunk 2 above)
Comment (grliszas14) [reply au4/src/projectscene/view/timeline/timeformat.h]: added

**Pull request merged**